### PR TITLE
confgenerator: add resource_name label using modify fields.

### PR DESCRIPTION
**Pull request by franciscovalentecastro**
b/169784211

### CODE REVIEW

File: confgenerator/logging.go [hunk 1]
```diff
@@ -21,12 +21,15 @@ import (
...
 
 // setLogNameComponents generates a series of components that rewrites the tag on log entries tagged `tag` to be `logName`.
-func setLogNameComponents(tag, logName, receiverType string) []fluentbit.Component {
+func setLogNameComponents(tag, logName, receiverType string, hostName string) []fluentbit.Component {
 	return LoggingProcessorModifyFields{
 		Fields: map[string]*ModifyField{
 			"logName": {
 				DefaultValue: &logName,
 			},
+			`labels."agent.googleapis.com/resource_name"`: {
```
Comment (quentinmit) [inline confgenerator/logging.go:30]: I think from reading the tea leaves (https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/dce3e4b53ec4ff35e72d3c5fac1cf38428048e9d/lib/fluent/plugin/common.rb#L23 and https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/dce3e4b53ec4ff35e72d3c5fac1cf38428048e9d/lib/fluent/plugin/out_google_cloud.rb#L1126) that this should actually be

```
compute.googleapis.com/resource_name
```

to match the legacy logging agent.

File: confgenerator/logging.go [hunk 2]
```diff
@@ -21,12 +21,15 @@ import (
...
+				StaticValue: &hostName,
```
Comment (quentinmit) [inline confgenerator/logging.go:31]: Use `DefaultValue` so that users can override it.

File: confgenerator/confgenerator.go [hunk 3]
```diff
@@ -209,7 +209,7 @@ func (l *Logging) generateFluentbitComponents(userAgent string, hostInfo *host.I
 					}
 					components = append(components, processor.Components(tag, strconv.Itoa(i))...)
 				}
-				components = append(components, setLogNameComponents(tag, rID, receiver.Type())...)
+				components = append(components, setLogNameComponents(tag, rID, receiver.Type(), hostInfo.Hostname)...)
```
Comment (quentinmit) [inline confgenerator/confgenerator.go:212]: hostname isn't necessarily the same as the instance name - the resource name has to come from the GCE metadata server

File: confgenerator/logging.go (hunk 1 above)
Comment (franciscovalentecastro) [reply confgenerator/logging.go]: That's right! Fixed.

File: confgenerator/logging.go (hunk 2 above)
Comment (franciscovalentecastro) [reply confgenerator/logging.go]: Done!

File: confgenerator/confgenerator.go (hunk 3 above)
Comment (franciscovalentecastro) [reply confgenerator/confgenerator.go:212]: I went back to explore the initial discussion ( b/169784211#comment18 ) with @igorpeshansky and @qingling128 about this task and realized why it was ok to use `hostInfo.Hostname`. 

The [value that is used](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/dce3e4b53ec4ff35e72d3c5fac1cf38428048e9d/lib/fluent/plugin/out_google_cloud.rb#L1126) in `google-fluentd` (which this feature wants to mirror) is actually the [`Socket.gethostname`](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/common.rb#L196
) which is equivalent of getting `os.Hostname` in go with `gopsutil`.

File: integration_test/ops_agent_test.go [hunk 4]
```diff
@@ -679,6 +679,50 @@ func TestModifyFields(t *testing.T) {
...
+		config := fmt.Sprintf(`logging:
+  receivers:
+    f1:
+      type: files
+      include_paths:
+      - %s
+  processors:
+    json:
+      type: parse_json
+  exporters:
```
Comment (qingling128) [inline integration_test/ops_agent_test.go:698]: [Optional] `exporters` are not required and can be inferred from the configuration if they are not specified.

**Review by qingling128 — APPROVED** [review]
LGTM

File: integration_test/ops_agent_test.go (hunk 4 above)
Comment (quentinmit) [reply integration_test/ops_agent_test.go:698]: They're not only not required, but they're completely ignored if specified.

**Pull request merged**